### PR TITLE
Allow QueueAdapter to locally dismiss items

### PR DESCRIPTION
**Pull request by udiWertheimer**
`QueueAdapter` (inheriting from `ParseQueryAdapter`) can now dismiss items locally. This prevents the synchronization problems that manifested in the "swipe to dismiss" animation.
Now, when swiping a queue item, it is first locally dismissed in the adapter, and only then is it removed remotely in Parse.

mokoanimate

### CODE REVIEW

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java [hunk 1]
```diff
@@ -49,6 +53,8 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
             public EnhancedListView.Undoable onDismiss(final EnhancedListView enhancedListView, int i) {
                 try {
                     new LeaveQueueOperation().run((QueueItem) mQueueListView.getItemAtPosition(i));
+                    mQueueAdapter.dismissItem(i);
+                    mQueueAdapter.notifyDataSetChanged();
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:56]: this updates parse?

**Comment by yonbergman** [comment]
cool

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java (hunk 1 above)
Comment (udiWertheimer) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:56]: nope. `LeaveQueueOperation` does.

see below for `dismissItem`. It does a sort of "local dismiss" to make the adapter act as if the dismissed item no longer exists (even though it does, until the adapter gets the changes from the server)

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java [hunk 2]
```diff
@@ -86,6 +93,39 @@ public QueueAdapter(Context context) {
...
+                @Override
+                public void onLoading() {
+                    // Must be implemented
+                }
+
+                @Override
+                public void onLoaded(List<QueueItem> queueItems, Exception e) {
+                    // Initialize the array of non-dismissed positions to include all queried items
+
+                    Integer[] positionsArray;
```
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java]: imho `int[]` will perform better than `Integer[]`

**Comment by giladgo** [comment]
:tada:

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java (hunk 1 above)
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:56]: but how is Leave queue called?

Comment (udiWertheimer) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:56]: it's right there, line 55, one line before the call to `dimissItem`

Comment (udiWertheimer) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:56]: it was already there from previous PR

**Pull request merged**